PerimeterXModule/PxModule.cs

```diff
@@ -317,21 +320,48 @@ public static void ResponseBlockPage(PxContext pxContext, PxModuleConfigurationS
             {
                 template = "challenge";
             }
-            else if (pxContext.BlockAction == "c")
-            {
-                template = config.CaptchaProvider;
-            }
-
-            Debug.WriteLine(string.Format("Using {0} template", template), PxConstants.LOG_CATEGORY);
-
-            // In the case of a challenge, the challenge response is taken directly from BlockData. Otherwise, generate html template.
-            content = template == "challenge" && !string.IsNullOrEmpty(pxContext.BlockData) ? pxContext.BlockData :
-                TemplateFactory.getTemplate(template, config, pxContext.UUID, pxContext.Vid);
-
-            pxContext.ApplicationContext.Response.Write(content);
-        }
-
-        public void Dispose()
+			else if (pxContext.BlockAction == "c")
+			{
+				template = config.CaptchaProvider;
+			}
+
+			if (pxContext.CookieOrigin.Equals(CookieOrigin.HEADER))
+			{
+				template = string.Format("{0}Mobile", template);
+			}
```

Comment:
It doesn't look good to have `if (pxContext.CookieOrigin.Equals(CookieOrigin.HEADER))` here for the `<template>Mobile` and then same `if` again for the actual response generation.
I think that you should move the upper if into the `getTemplate` code (pass the `pxContext.CookieOrigin` as a param) as it is only relevant for the template generation logic.